### PR TITLE
ops: kokkos: revise `clone`

### PR DESCRIPTION
**Pull request by mzuzek**
refs #517

### Overloads

Kokkos `clone` overloads:

| function | input `T` |
|:---:|:---:|
| `clone( T )` | native Kokkos vector or matrix (no expressions) |

### Tests

Unit tests (in `tests/functional_small/ops/`):

| file name | test name | tested type |
|:---|:---|:---:|
| `ops_kokkos_vector.cc` | `ops_kokkos.vector_clone` | `Kokkos::View<double*>` |
| `ops_kokkos_matrix.cc` | `ops_kokkos.dense_matrix_clone` | `Kokkos::View<double**>` |

### CODE REVIEW

**Review by fnrizzi — CHANGES_REQUESTED** [review]
please *remove* more all overloads that operate on expressions. 
For now i don't want to clone an expression. Before we do this, we need to think about what semantics we want that to have.

File: include/pressio/ops/kokkos/ops_clone.hpp [hunk 1]
```diff
@@ -53,30 +53,20 @@ namespace pressio{ namespace ops{
...
-  or ::pressio::is_expression_acting_on_kokkos<T>::value)
-  and ::pressio::Traits<T>::rank == 2,
-  T
-  >
-clone(const T & clonable)
-{
-  T r(clonable.label(), clonable.extent(0), clonable.extent(1));
-  Kokkos::deep_copy(r, clonable);
-  return r;
+  auto mirror = Kokkos::create_mirror(clonable);
```
Comment (fnrizzi) [inline include/pressio/ops/kokkos/ops_clone.hpp:67]: @mzuzek why did you change the implementation? 
This is not correct: create_mirror does not clone anything, it is a clone just because you are doing things on host. Please use the implementation I had before, maybe with the minor fix for the label:
```
  T r(clonable.label()+"_clone", clonable.extent(0));
  Kokkos::deep_copy(r, clonable);
  return r;
```

**Review by fnrizzi — CHANGES_REQUESTED** [review]
see my comment

File: include/pressio/ops/kokkos/ops_clone.hpp (hunk 1 above)
Comment (mzuzek) [reply include/pressio/ops/kokkos/ops_clone.hpp]: thanks, fixed on 07f5020901b5c7cfbb60039cdfd95c84b1b2cf42

**Review by fnrizzi — APPROVED** [review]
